spec/controllers/answers_controller_spec.rb

```diff
@@ -119,4 +120,70 @@
       end
     end
   end
+
+  describe 'PATH #update' do
+    let!(:answer) { create(:answer, question: question, user: author) }
+
+    context 'with valid attributes' do
+      before { login(author) }
+
+      it 'change answer attributes' do
+        patch :update, params: { id: answer, answer: { body: 'new body' }, question_id: answer.question }, format: :js
+        answer.reload
+        expect(answer.body).to eq 'new body'
+      end
+
+      it 'renders update view' do
+        patch :update, params: { id: answer, answer: { body: 'new body' }, question_id: answer.question }, format: :js
+        expect(response).to render_template :update
+      end
+    end
+
+    context 'with invalid attributes' do
+      before { login(author) }
+
+      it 'can not change attributes' do
+        expect do
+          patch :update, params: { id: answer, answer: attributes_for(:answer, :invalid), question_id: answer.question }, format: :js
+        end.to_not change(answer, :body)
+      end
+
+      it 'renders update view' do
+        patch :update, params: { id: answer, answer: attributes_for(:answer, :invalid), question_id: answer.question }, format: :js
+        expect(response).to render_template :update
+      end
+    end
+  end
+
+  describe 'PATCH #best' do
+    let!(:answer) { create(:answer, question: question, user: author) }
+    let!(:answer_somebody) { create(:answer, question: question_somebody, user: user) }
+    before { login(author) }
+
+    context 'as auhtor of question' do
+      it 'can change attribute' do
+        patch :best, params: { id: answer, question_id: answer.question }, format: :js
+        answer.reload
+        expect(answer.best).to eq true
+      end
+
+      it 'renders update view' do
+        patch :best, params: { id: answer, question_id: answer.question }, format: :js
+        expect(response).to render_template :best
+      end
+    end
+
+    context 'as not auhtor of question' do
+      it 'can not change attribute' do
+        patch :best, params: { id: answer_somebody, question_id: answer_somebody.question }, format: :js
+        answer.reload
+        expect(answer.best).to eq false
```

Comment:
https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/predicate-matchers